### PR TITLE
Dynamic form fields based on choice of attribute-group field (attribute form)

### LINKED ISSUE #36010: related #35591 -bug 3 :  Attribute value list doesn't adapt itself if the attribute type changes
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Regression on the migrated Attributes pages

see #35591 B3 :  Attribute value list doesn't adapt itself if the attribute type changes

### Expected behavior

_No response_

### Steps to reproduce

When I add a new value and change the attribute group, the inputs for that attribute are not displayed. In my case, I have to go to my Color attribute and edit it to add a color manually.
Steps to reproduce :

Go to BO > Attributes page > View Paper Type
Click Add new value button
In Attribute group > Change it to Color
The inputs for the color attributes are not automatically displayed ❌
Same case the other way around.

Go to BO > Attributes page > View Color
Click Add new value button
In Attribute group > Change it to Dimension
The inputs for the color attributes are not automatically removed ❌

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by mattgoud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | related #35591 - bug 3 : Attribute value list doesn't adapt itself if the attribute type changes
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Indicate how to verify that this change works as expected.
| UI Tests          | Please run UI tests and paste here the link to the run. [Read this page to know why and how to use this tool.](https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/ui-tests/).
| Fixed issue or discussion?     | Fixes #36010 
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by mattgoud** [comment]
sorry @matthieu-rolland I pushed a new commit (refacto js) which canceled your review, if you could take a look thanks

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php [hunk 1]
```diff
@@ -106,17 +106,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'help' => $this->trans('Your internal name for this attribute.', 'Admin.Catalog.Help')
                     . '&nbsp;' . $this->trans('Invalid characters:', 'Admin.Notifications.Info')
                     . ' ' . TypedRegexValidator::CATALOG_CHARS,
-            ]);
-
-        if ($hasAttributeGroupId === true && $attributeGroup->group_type === AttributeGroupType::ATTRIBUTE_GROUP_TYPE_COLOR) {
```
Comment (nicosomb) [inline src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php:111]: I added this check in my recent PR (https://github.com/PrestaShop/PrestaShop/pull/35865/files#diff-010f6eb7f9b60c2911e7db12103d0575cdacdcb29f4da24cfb67be0bbf1fd996R111). Are you sure that we can remove it?

**Review by nicosomb — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php:111]: I think it's required if we want to toggle the fields dynamically

**Review by jolelievre — APPROVED** [review]
Thanks @mattgoud

**Review by nicosomb — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @mattgoud,

I tested your PR and it's good for me! :tada: 

Thanks!

https://www.loom.com/share/bb66bb3f3f064a6286e7e0c3ac76a223?sid=82dd0a79-9dec-463a-89cf-ba6226c4ac75

**Review by paulnoelcholot — APPROVED** [review]

**Comment by kpodemski** [comment]
thank you @mattgoud

**Comment by mattgoud** [comment]
> thank you @mattgoud

great ! thank you @kpodemski

**Pull request merged**